staging/src/k8s.io/cloud-provider/controllers/service/controller.go

```diff
@@ -697,6 +706,14 @@ func shouldSyncUpdatedNode(oldNode, newNode *v1.Node) bool {
 	if oldNode.Spec.ProviderID != newNode.Spec.ProviderID {
 		return true
 	}
+	// can happen only at the beginning of the node lifecycle
```

Comment:
I don't think these comments, as written, add a lot of value. We might be better off without them, alternatively mention that we explicitly want to trigger a sync loop when these annotations change.

Reply:
removed